connectors/sources/sharepoint_server.py

```diff
@@ -414,51 +333,22 @@ async def _api_call(self, url_name, url="", **url_kwargs):
 
         while True:
             try:
-                if self.is_cloud:
-                    await self._set_access_token()
-                    headers = {"Authorization": f"Bearer {self.access_token}"}
                 async with self._get_session().get(  # pyright: ignore
                     url=url,
-                    ssl=self.ssl_ctx,
+                    ssl=self.ssl_ctx,  # pyright: ignore
                     headers=headers,
                 ) as result:
                     if url_name == ATTACHMENT:
                         yield result
                     else:
                         yield await result.json()
                     break
-            except ClientResponseError as exception:
-                retry, retry_seconds = self._get_retry_after(
-                    retry=retry, exception=exception
-                )
-                response_headers = exception.headers or {}
-                if exception.status == 429:
-                    if "Retry-After" in response_headers:
-                        retry_seconds = int(response_headers["Retry-After"])
-                    else:
-                        logger.warning(
-                            "Response Code from Sharepoint Server is 429 but Retry-After header is not found"
-                        )
-                        retry_seconds = DEFAULT_RETRY_SECONDS
-                    logger.warning(
-                        f"Rate Limited by Sharepoint: have {response_headers['RateLimit-Remaining']} of {response_headers['RateLimit-Limit']} left, opportunity to retry in {retry_seconds} seconds"
-                    )
-                elif "token has expired" in response_headers.get(  # pyright: ignore
-                    "x-ms-diagnostics", ""
-                ):
-                    await self._set_access_token()
-                logger.debug(f"Will retry in {retry_seconds} seconds")
-                await self._sleeps.sleep(retry_seconds)
-
-            except ServerDisconnectedError as exception:
-                retry, retry_seconds = self._get_retry_after(
-                    retry=retry, exception=exception
-                )
-                await self.close_session()
-                logger.debug(f"Will retry in {retry_seconds} seconds")
-                await self._sleeps.sleep(retry_seconds)
-
             except Exception as exception:
+                if isinstance(
+                    exception,
+                    ServerDisconnectedError,
```

Comment:
No retrying after logic here?

Reply:
Yes retry is there on line 352 we have just wrap the complete retry in a separate method.